File: javascriptChallenges/fibonacci.js
function nextFibNum(arr){
  return arr[arr.length - 2] + arr[arr.length - 1]
}

module.exports = function fibonacci(num){
  if(num <= 2){
    return [0, 1]
  } else if(num > 2){
    var fibArray = fibonacci(num - 1)
    var nextFib  = nextFibNum(fibArray)
    fibArray.push(nextFib)
    return fibArray
  }
}

File: Ruby/lib/fibonacci.rb
require 'pry-state'
class Fibonacci

  def sequence(num)
    fib_array = [0, 1]
    fib_array if fib_array.length == num
    fib_array = construct_sequence(num, fib_array)
    fib_array
  end

  def construct_sequence(num, fibonacci)
    (num - 2).times do
      fibonacci.push(fibonacci[-2] + fibonacci[-1])
    end
    fibonacci
  end

end
